File: app/src/main/java/me/imzack/app/end/receiver/BootCompleteReceiver.java
package me.imzack.app.end.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import me.imzack.app.end.App;
import me.imzack.app.end.util.SystemUtil;

import java.util.Map;

public class BootCompleteReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        Map<String, Long> reminderTimeMap = App.getDataManager().getDatabaseHelper().queryReminderTimeWithEnabledReminder();
        for (Map.Entry<String, Long> entry : reminderTimeMap.entrySet()) {
            SystemUtil.setReminder(entry.getKey(), entry.getValue());
        }
    }
}

File: app/src/main/kotlin/me/imzack/app/end/receiver/BootCompleteReceiver.kt
package me.imzack.app.end.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import me.imzack.app.end.model.DataManager
import me.imzack.app.end.util.SystemUtil

class BootCompleteReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val reminderTimeMap = DataManager.databaseHelper.queryReminderTimeWithEnabledReminder()
        for ((key, value) in reminderTimeMap) {
            SystemUtil.setReminder(key, value)
        }
    }
}
